### DISCUSSION TITLE
when does the getInitialProps of _app run?

### DISCUSSION THREAD

**Discussion by tylerthehaas** [vercel/next.js#11137] [Help]
I have an app that is using _app.js getInitialProps to fetch user info to render inside of a header. That works good on the first render of the app however if I logout of that user and login with a different user getInitialProps isn't run again so the previous users data is rendered. This seems like there is some sort of caching strategy with the custom app page that I haven't been able to find documented anywhere. Anyone know why this behavior is occuring?

**Comment by shahruz** [comment]
I've been dealing with a similar issue, deploying on Now / Vercel. 

My _app.js's getInitialProps checks for a cookie, and if it's there it requests basic user information from an API, then passes that data as a prop to the App component. 

After deploying, the first page load seems to cache the data from getInitialProps the first time it's run for any specific page, so if I load the home page while logged in, then open an incognito window and open the home page, it'll use my user info from the other window for a second on load. Based on the logs in the API call the user information isn't being re-requested on each load, meaning _app.js's initialProps are being cached somewhere by default.

**Reply by V01dgazer** [reply]
Have you managed to solve the issue? I have the same problem.

**Reply by shahruz** [reply]
I didn't find a solution unfortunately. I ended up adjusting the architecture to not use getInitialProps on _app.js.
